solr/core/src/java/org/apache/solr/search/join/BlockJoinParentQParser.java

```diff
@@ -60,10 +60,15 @@ protected String getFiltersParamName() {
   }
 
   protected Query parseParentFilter() throws SyntaxError {
-    String filter = localParams.get(getParentFilterLocalParamName());
-    QParser parentParser = subQuery(filter, null);
-    Query parentQ = parentParser.getQuery();
-    return parentQ;
+    try {
```

Comment:
Why; what exception gets thrown before your change?  If I recall, this whole method would return null and not throw; right?  So check that parentQ isn't null and if it is, throw a SyntaxError.